public/app/plugins/datasource/cloud-monitoring/specs/datasource.test.ts

```diff
@@ -213,51 +213,6 @@ describe('CloudMonitoringDataSource', () => {
       });
     });
   });
-
-  describe('unit parsing', () => {
-    const { ds } = getTestcontext();
-
-    describe('when theres only one target', () => {
-      describe('and the cloud monitoring unit does nott have a corresponding grafana unit', () => {
-        it('should return undefined', () => {
-          const res = ds.resolvePanelUnitFromTargets([{ unit: 'megaseconds' }]);
-
-          expect(res).toBeUndefined();
-        });
-      });
-      describe('and the cloud monitoring unit has a corresponding grafana unit', () => {
-        it('should return bits', () => {
-          const res = ds.resolvePanelUnitFromTargets([{ unit: 'bit' }]);
-
-          expect(res).toEqual('bits');
-        });
-      });
-    });
-
-    describe('when theres more than one target', () => {
-      describe('and all target units are the same', () => {
-        it('should return bits', () => {
-          const res = ds.resolvePanelUnitFromTargets([{ unit: 'bit' }, { unit: 'bit' }]);
-
-          expect(res).toEqual('bits');
-        });
-      });
-      describe('and all target units are the same but does not have grafana mappings', () => {
-        it('should return the default value of undefined', () => {
-          const res = ds.resolvePanelUnitFromTargets([{ unit: 'megaseconds' }, { unit: 'megaseconds' }]);
-
-          expect(res).toBeUndefined();
-        });
-      });
-      describe('and all target units are not the same', () => {
-        it('should return the default value of undefined', () => {
-          const res = ds.resolvePanelUnitFromTargets([{ unit: 'bit' }, { unit: 'min' }]);
-
-          expect(res).toBeUndefined();
-        });
-      });
-    });
-  });
```

Comment:
Have these tests moved to the backend now?

Reply:
I moved them here cloudmonitoring_test.go, but since now we have one unit per target(dataframe), i just check whether there is a map gcp unit -> grafana unit, if there is i put the unit in dataframe.config, otherwise it is empty.

Reply:
Ok, thank you.

Reply:
hi hugo, I just applied all logics into backend with all the test removed above.